File: server/src/main/java/com/springboot/store/entity/Customer.java
package com.springboot.store.entity;

import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.*;
import lombok.*;

import java.util.Date;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder

@Entity
@Table(name = "customers", uniqueConstraints = {
        @UniqueConstraint(name = "customer_email_unique", columnNames = {"email"})
})
public class Customer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "email", nullable = false)
    private String email;

    @Column(name = "phone_number")
    private String phoneNumber;

    @Column(name = "address")
    private String address;

    @Column(name = "sex")
    private String sex;

    @Column(name = "description")
    private String description;

    @Column(name = "status")
    private String status;

    @Column(name = "birthday")
    @JsonFormat(pattern="yyyy-MM-dd")
    private Date birthday;

    @Column(name = "created_at")
    @JsonFormat(pattern="yyyy-MM-dd HH:mm:ss")
    private Date createdAt;

    @ManyToOne()
    @JoinColumn(name = "customer_group_id")
    private CustomerGroup customerGroup;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "creator_id")
    private Staff creator;

}

File: entities/Customer.ts
export type Customer = {
  id: number;
  name: string;
  customerGroup: string;
  phoneNumber: string;
  address: string;
  sex: "Male" | "Female" | "Not to say";
  email: string;
  birthday: string;
  creatorId: number;
  createdAt: string;
  description: string;
  image: {
    id: number,
    url: string,
  } | null;
  isDeleted: boolean;
};

export type CustomerGroup = {
  id: number;
  name: string;
  createdDate: string;
};
